### DISCUSSION TITLE
Aqara curtain robot cannot be upgraded

### DISCUSSION THREAD

**Discussion by eigenphase** [Koenkk/zigbee2mqtt#18651] [Devices]
I have two [ZNCLBL01LM](https://www.zigbee2mqtt.io/devices/ZNCLBL01LM.html) and z2m says they have an OTA upgrade. However, I am not able to upgrade the firmware. Below how it looks in the webinterface:

<img width="1161" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/7464149/2729123a-f282-42e6-96e0-008c2a6dc4e8">

When I click Update device firmware for the first one, it says update is starting but then nothing happens.

When I click on the second one I just get:

<img width="301" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/7464149/7d136c6a-ad0f-45e1-bc9a-67fa8567108a">

How can I upgrade these firmwares?

**Comment by Koenkk** [comment]
Fixed!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Reply by eigenphase** [reply]
Absolutely awesome thank you!!

But what am I doing wrong when switching to dev branch?

```
root@rfpi:/opt/zigbee2mqtt# sudo -u zigbee2mqtt git fetch origin dev
From https://github.com/Koenkk/zigbee2mqtt
 * branch            dev        -> FETCH_HEAD
root@rfpi:/opt/zigbee2mqtt# sudo -u zigbee2mqtt git checkout dev
error: pathspec 'dev' did not match any file(s) known to git
root@rfpi:/opt/zigbee2mqtt#
```

**Reply by cristiaandr** [reply]
@Koenkk this issue seems to be still present. Any chances on reopening this one? Cheers

**Comment by kiekerjan18** [comment]
Sadly, this seem not to be fixed in 1.33.0 as indicated in changelog. I have two of these devices not upgrading

![image](https://github.com/Koenkk/zigbee2mqtt/assets/63964079/b617a460-0b9c-4150-af9b-a9ee1ace9118)

**Comment by mocrie78** [comment]
Z2M Version 1.33.0-1. Same experience as kiekerjan18 - unfortunately the problem does not appear to be fixed. The Xiaomi devices (current firmware version 0.0.0_0027 ) are not upgrading. Same failure message "Timeout: device did not request any image blocks".

**Comment by eigenphase** [comment]
Same for me, unfortunately. One thing that really confuses me is that the identical device shows up differently: Once given current version, once it's just blank:

<img width="1130" alt="image" src="https://github.com/Koenkk/zigbee2mqtt/assets/7464149/215e90fd-7f6f-4b43-b340-cc643b63b23d">

No update possible for either :-( When updating I now get the same error messages as @kiekerjan18

**Comment by rajil** [comment]
I am using 1.33.0 and not able to update the firmware.
```
zigbee2mqtt  | Zigbee2MQTT:info  2023-10-01 11:19:26: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/update', payload '{"data":{"id":"Bedroom_Curtain_Right"},"error":"Update of 'Bedroom_Curtain_Right' failed (Timeout: device did not request any image blocks)","status":"error","transaction":"6o9io-3"}'
zigbee2mqtt  | Zigbee2MQTT:error 2023-10-01 11:19:26: Update of 'Bedroom_Curtain_Right' failed (Timeout: device did not request any image blocks)
```
